src/app/pages/storage/volumes/permissions/stores/dataset-acl-editor.store.ts

```diff
@@ -312,54 +313,102 @@ export class DatasetAclEditorStore extends ComponentStore<DatasetAclEditorState>
+            catchError((error) => {
+              new EntityUtils().errorReport(error, this.dialog);
+              markAceAsHavingErrors(index);
+              return EMPTY;
+            }),
+          ),
         );
+
+        return;
       }
+
       if ([NfsAclTag.UserGroup, PosixAclTag.Group].includes(ace.tag)) {
-        return this.userService.getGroupByName(ace.who).pipe(
-          map((group) => ({ ...aceAttributes, id: group.gr_gid })),
-          catchError((error) => {
-            new EntityUtils().errorReport(error, this.dialog);
-            markAceAsHavingErrors(index);
-            return of(aceAttributes);
-          }),
+        requests.push(
+          this.userService.getGroupByName(ace.who).pipe(
+            tap((group) => groupWhoToIds.set(ace.who, group.gr_gid)),
+            catchError((error) => {
+              new EntityUtils().errorReport(error, this.dialog);
+              markAceAsHavingErrors(index);
+              return EMPTY;
+            }),
+          ),
         );
       }
-
-      return of({
-        ...aceAttributes,
-        id: -1, // -1 is effectively null for middleware
-      });
     });
 
-    return forkJoin(prepareAces).pipe(
+    requests.push(
+      this.userService.getUserByName(options.owner).pipe(
+        tap((user) => userWhoToIds.set(options.owner, user.pw_uid)),
+        catchError((error) => {
+          new EntityUtils().errorReport(error, this.dialog);
+          return EMPTY;
+        }),
+      ),
+    );
+
+    requests.push(
+      this.userService.getGroupByName(options.ownerGroup).pipe(
+        tap((group) => groupWhoToIds.set(options.ownerGroup, group.gr_gid)),
+        catchError((error) => {
+          new EntityUtils().errorReport(error, this.dialog);
+          return EMPTY;
+        }),
+      ),
+    );
+
+    return forkJoin(requests).pipe(
       withLatestFrom(this.state$),
       filter(([_, currentState]) => currentState.acesWithError.length === 0),
-      map(([convertedAces]) => ({
-        options,
-        acltype: editorState.acl.acltype,
-        gid: null,
-        uid: null,
-        path: editorState.mountpoint,
-        dacl: convertedAces as NfsAclItem[] | PosixAclItem[],
-      } as SetAcl)),
+      map(([_, currentState]) => {
+        const convertedAces = (currentState.acl.acl as unknown[]).map((ace: NfsAclItem | PosixAclItem) => {
+          const aceAttributes = omit(ace, ['who']);
+          if ([NfsAclTag.User, PosixAclTag.User].includes(ace.tag)) {
+            const id = userWhoToIds.has(ace.who) ? userWhoToIds.get(ace.who) : -1;
+            return { ...aceAttributes, id };
+          }
+          if ([NfsAclTag.UserGroup, PosixAclTag.Group].includes(ace.tag)) {
+            const id = groupWhoToIds.has(ace.who) ? groupWhoToIds.get(ace.who) : -1;
+            return { ...aceAttributes, id };
+          }
+
+          return {
+            ...aceAttributes,
+            id: -1, // -1 is effectively null for middleware
```

Comment:
Is there a specific reason why we're sending `-1` instead of `null` here? If not, it just looks like it'll add unnecessary confusion. If we mean to send `null`, we should just send `null`

Reply:
Unfortunately `null` is not supported :)